Source/RapyutaSimulationPlugins/Private/Core/RRBaseActor.cpp

```diff
@@ -43,19 +42,26 @@ void ARRBaseActor::PreInitializeComponents()
 {
     Super::PreInitializeComponents();
 
-    // SETUP + CONFIGURE ESSENTIAL GAME & SIM COMMON OBJECTS
+    // SETUP + CONFIGURE GAME & SIM COMMON OBJECTS
     // NOTE: These are used ONLY for dynamic runtime actors.
     // In child BP actors, except for GameSingleton, other Game framework entities maybe not available yet
-    GameMode = URRCoreUtils::GetGameMode<ARRGameMode>(this);
-    GameState = URRCoreUtils::GetGameState<ARRGameState>(this);
+#if RAPYUTA_USE_SCENE_DIRECTOR
+    RRGameState = URRCoreUtils::GetGameState<ARRGameState>(this);
+    check(RRGameState)
+#endif
```

Comment:
```suggestion
    RRGameState = URRCoreUtils::GetGameState<ARRGameState>(this);
```
for still being used [here](https://github.com/rapyuta-robotics/RapyutaSimulationInternal/blob/devel/Source/RapyutaSimInternal/Private/Warehouse/RRPallet.cpp#L20)
So basically all of them are still in use

Reply:
if those are used only in one child class. I think it is make sense to move it to that class instead of keeping it in base class.

Reply:
So our RRGame~ actors could be used anywhere in our Sim, not particularly a mesh or non-mesh actor, so the essential idea is to avoid duplication in child classes, which is the very reason why we put it here in the base one.

Reply:
I do not fully agreed but reverted since that is not main topic of this pr.

Reply:
Yp, let's do it step by step. My thinking is any refactoring should come with valid specific reasons, otherwise, it might raise some questions to future maintainers on the consistency of our large-growing source base.
* This PR is meant to reduce the coupling with RRGame~ meaning `check` should not be imposed, meaning the use of `RRGame~` is optional (as up to users to decide it should be used or not).
* There was also no mention of the rationale in the PR description that member pointers should be also removed, which would cause an impact to client source.